### DISCUSSION TITLE
is 1.18 experiemental supported

### DISCUSSION THREAD

**Discussion by lifewater** [itzg/docker-minecraft-server#990] [Q&A]
Hey, first time using this docker image.  Is it possible to run 1.18 experimentals from this?  I tried building a docker compose for it but i got the following error:

mc_1  | [init] ERROR couldn't find a matching manifest entry for 1.18_experimental-snapshot-2
I tested 1.17.1 just to make sure and that worked without issues

**my docker-compose.yaml:**
version: "3"

services:
  mc:
    image: itzg/minecraft-server
    ports:
      - 25565:25565
    environment:
      #TYPE: "PAPER"
      EULA: "TRUE"
      VERSION: "1.18_experimental-snapshot-2"
    volumes:
      - /home/lifewater/minecraft/data:/data

(i got the version format from MultiMC, unsure if its correct)

**Comment by itzg** [comment]
Since Mojang is doing these odd 1.18 pre-snapshot releases outside their normal mechanism it is not seamlessly supported like every other snapshot version used to be. I haven't tried yet, but using TYPE=CUSTOM and supplying the pre-downloaded server jar should work.

**Reply by lifewater** [reply]
Edit:  I thought it worked, but for some reason it thinks that its snapshot 1, when in fact the jar file is snapshot 2. 

When starting up the container, i see the following:
mc_1  | [21:37:42] [Worker-Main-10/INFO]: Loaded 7 recipes
mc_1  | [21:37:42] [Worker-Main-10/INFO]: Loaded 1138 advancements
mc_1  | [21:37:44] [Server thread/INFO]: Starting minecraft server version 1.18 Experimental Snapshot 1

When running it via java -> java -Xmx2048m -Xms2048m -jar server.jar, i see the following
[16:42:27] [Worker-Main-12/INFO]: Loaded 7 recipes
[16:42:27] [Worker-Main-12/INFO]: Loaded 1138 advancements
[16:42:29] [Server thread/INFO]: Starting minecraft server version 1.18 experimental snapshot 2

Any idea what I'm doing wrong here?  This is my docker-compose

version: "3"

services:
  mc:
    image: itzg/minecraft-server
    ports:
      - 25565:25565
    environment:
      TYPE: "CUSTOM"
      EULA: "TRUE"
      CUSTOM_SERVER: "https://launcher.mojang.com/v1/objects/83a3c2e94b744ad8f0c8bc373dd70a85da59babf/server.jar"
    volumes:
      - ./minecraft/data:/data

**Reply by finndo77** [reply]
testing now, will update this in a few, I have a couple of different things going on, so it might be an hour.  to your initial post, I wanted to say that the version is currently using underscores, not periods...
1_18_experimental-snapshot-5

but I am going to try and get the snapshot 5 running

edit 1:
I pulled this URL out of the JSON file from the official minecraft.net post about exp snap 5
https://launcher.mojang.com/v1/objects/6680fe47a8d599193219f155d7d878feb05bfeb3/server.jar

edit 2: my docker launch command and docker log

I don't use docker compose, so this is a simple shell script. I have no idea why not, I think I had an issue either getting it to work or setting it up or something for ever ago and gave up trying as at the time, one minecraft server was the only thing I was using docker for and I just never bothered to do anything different than what was working already.

```
#! /bin/bash
echo "starting up Snapshot server"

docker run -d -e INIT_MEMORY=2G -e MAX_MEMORY=5G -e PVP=FALSE -e SPAWN_PROTECTION=TRUE -e VIEW_DISTANCE=18 -e USE_AIKAR_FLAGS=true -e MOTD="Minecraft 1.18 Experimental Snapshot Server" -e LEVEL=exp-snapshot -e DIFFICULTY=normal -e EULA=TRUE -e MAX_WORLD_SIZE=90000 -e FORCE_GAMEMODE=TRUE -e SERVER_NAME=exp-snapshot -e TYPE=CUSTOM -e CUSTOM_SERVER="https://launcher.mojang.com/v1/objects/6680fe47a8d599193219f155d7d878feb05bfeb3/server.jar" -p 45579:25565 -e ENABLE_AUTOPAUSE=FALSE -v /mnt/minecraft-2/exp-snapshot:/data -v /etc/timezone:/etc/timezone:ro --name mc-exp-snapshot itzg/minecraft-server:java16

exit
```


here is the docker log for the server starting up, seems good, I won't be able to log on for another 30 mins or so

```
[init] Changing ownership of /data to 1000 ...
[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x 2 1000 1000 4096 Aug 29 01:16 /data'
[init] Resolved version given LATEST into 1.17.1
[init] Resolving type given CUSTOM
[init] Downloading custom server jar from https://launcher.mojang.com/v1/objects/6680fe47a8d599193219f155d7d878feb05bfeb3/server.jar ...
[init] Creating server.properties in /data/server.properties
[init] Disabling whitelist
[init] Setting whitelist to 'false' in /data/server.properties
[init] Setting white-list to 'false' in /data/server.properties
[init] Adding server-name with 'exp-snapshot' in /data/server.properties
[init] Setting server-port to '25565' in /data/server.properties
[init] Setting motd to 'Minecraft 1.18 Experimental Snapshot Server' in /data/server.properties
[init] Setting spawn-protection to 'true' in /data/server.properties
[init] Setting view-distance to '18' in /data/server.properties
[init] Setting max-build-height to '320' in /data/server.properties
[init] Setting force-gamemode to 'true' in /data/server.properties
[init] Setting enable-rcon to 'true' in /data/server.properties
[init] Setting rcon.password to 'minecraft' in /data/server.properties
[init] Setting rcon.port to '25575' in /data/server.properties
[init] Setting max-world-size to '90000' in /data/server.properties
[init] Setting level-name to 'exp-snapshot' in /data/server.properties
[init] Setting pvp to 'false' in /data/server.properties
[init] Setting online-mode to 'true' in /data/server.properties
[init] Setting difficulty to 'normal' in /data/server.properties
[init] Updating ops
[init] Checking for JSON files.
[init] Setting initial memory to 2G and max to 5G
[init] Using Aikar's flags
[init] Starting the Minecraft server...
[21:17:03] [main/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[21:17:03] [main/WARN]: Unable to create a jar-filesystem for: jar:file:/data/server.jar!/assets/.mcassetsroot: java.nio.file.FileSystemNotFoundException
[21:17:04] [main/WARN]: Ambiguity between arguments [teleport, location] and [teleport, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
[21:17:04] [main/WARN]: Ambiguity between arguments [teleport, location] and [teleport, targets] with inputs: [0.1 -0.5 .9, 0 0 0]
[21:17:04] [main/WARN]: Ambiguity between arguments [teleport, destination] and [teleport, targets] with inputs: [Player, 0123, @e, dd12be42-52a9-4a91-a8a1-11c01849e498]
[21:17:04] [main/WARN]: Ambiguity between arguments [teleport, targets] and [teleport, destination] with inputs: [Player, 0123, dd12be42-52a9-4a91-a8a1-11c01849e498]
[21:17:04] [main/WARN]: Ambiguity between arguments [teleport, targets, location] and [teleport, targets, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
[21:17:04] [main/INFO]: Reloading ResourceManager: Default
[21:17:04] [Worker-Main-14/INFO]: Loaded 7 recipes
[21:17:04] [Worker-Main-14/INFO]: Loaded 1138 advancements
[21:17:06] [Server thread/INFO]: Starting minecraft server version 1.18 experimental snapshot 5
[21:17:06] [Server thread/INFO]: Loading properties
[21:17:06] [Server thread/INFO]: Default game type: SURVIVAL
[21:17:06] [Server thread/INFO]: Generating keypair
[21:17:06] [Server thread/INFO]: Starting Minecraft server on *:25565
[21:17:06] [Server thread/INFO]: Using epoll channel type
[21:17:06] [Server thread/INFO]: Preparing level "exp-snapshot"
[21:17:07] [Server thread/WARN]: Unable to find spawn biome
[21:17:19] [Server thread/INFO]: Preparing start region for dimension minecraft:overworld
```

and then it generated spawn, very very slowly

**Reply by finndo77** [reply]
server loaded in just fine without issue

**Reply by itzg** [reply]
Excellent. FYI for future pre-snapshot-snapshots, the wiki page arrived at the same server jar you pulled out of the json

https://minecraft.fandom.com/wiki/Java_Edition_1.18_experimental_snapshot_5

**Comment by finndo77** [comment]
@itzg everything seems to be working using custom type and providing the DL link to the server.jar.

I think you can close this one out with the answer of "yes"

**Reply by keepsmilyn** [reply]
None of the DL links mentioned above work for me. 

What exact environment variables have you changed / used to enable it to work with higher than 1.17.x? No settings I have adjusted have enabled it to function with anything higher. Container in docker continually restarts.

**Reply by itzg** [reply]
What exact environment variables have you tried? What image are you using?

**Reply by keepsmilyn** [reply]
Image that was being used:  itzg / minecraft-server:latest

Attached is the server.properties file as I no longer have the container installed. All environment variables used worked with 1.17.x, just not any later versions. I referenced server.properties from here for the Java edition: https://minecraft.fandom.com/wiki/Server.properties

[server_properties.txt](https://github.com/itzg/docker-minecraft-server/files/8405865/server_properties.txt)

**Comment by keepsmilyn** [comment]
Doesnt work for me on anything above 1.17.1. :-(

**Comment by finndo77** [comment]
Did you post in the right issue?

The wiki site has the download urls to use for the exp snaps.  Considering
nearly every single minecraft version is available from the launcher, I
don't think they got rid of the exps
